lib/internal/modules/esm/resolve.js

```diff
@@ -1021,8 +1090,20 @@ module.exports = {
   getPackageScopeConfig,
   getPackageType,
   packageExportsResolve,
-  packageImportsResolve
+  packageImportsResolve,
 };
 
 // cycle
 const { defaultGetFormat } = require('internal/modules/esm/get_format');
+if (policy) {
+  const $defaultResolve = defaultResolve;
+  module.exports.defaultResolve = async function defaultResolve(
+    specifier, context
+  ) {
+    const ret = await $defaultResolve(specifier, context, $defaultResolve);
+    policy.manifest.mightAllow(ret.url, () =>
+      new ERR_MANIFEST_DEPENDENCY_MISSING(context.url, specifier,
+                                          context.conditions));
+    return ret;
+  };
```

Comment:
What is this extra policy check accomplishing?

Reply:
this prevents network requests from firing if the resolution would be banned. network requests can extract data by doing things like putting secrets in query params

Reply:
Can you perhaps add a comment to this effect?